### PR TITLE
feat(conductor): implement verification of chain IDs commitment

### LINKED ISSUE #308: commitment and verification of namespaces in `SequencerNamespaceData`
`SequencerNamespaceData`, which is written to DA by the relayer, is as follows:

```
pub struct SequencerNamespaceData {
    pub block_hash: Hash,
    pub header: Header,
    pub last_commit: Option<Commit>,
    pub rollup_namespaces: Vec<Namespace>,
}
```

however, `rollup_namespaces` is not committed to during the consensus process, so the relayer can omit a namespace entirely without it being detectable by the conductor.

there should be a commitment of the rollup chain IDs in a block during the consensus process, allowing for verification that there was no data for a rollup completed omitted.

### PR DESCRIPTION
**Pull request by noot**
## Summary
previous PR implemented generating the chain IDs commitment in `prepare_proposal` and verifying it in `process_proposal`. this PR finishes the job by ensuring the chain IDs commitment is verified in conductor.

## Background
required so that we can ensure that the sequencer-relayer actually wrote data for every rollup ID in a block, previously it would have been able to omit data for a whole rollup without detection. we still need to implement slashing (eventually) for this - what happens right now is that the conductor will fail to verify a block if the commitment does not match up, even if it was able to obtain the rollup data for its own chain ID.

## Changes
-  ensuring the chain IDs commitment is verified in conductor
- update the sequencer app and `SequencerBlockData::from_tendermint_block` to handle a block with 2 commitments at the start of the txs correctly; this should have been done in my previous PR, apologies
- update ChainId to be max 32 bytes

## Testing
unit tests

## Related Issues

closes #308

### CODE REVIEW

File: crates/astria-conductor/Cargo.toml
Comment (SuperFluffy) [inline crates/astria-conductor/Cargo.toml:38]: I have started writing these like so:
```toml
sequencer-validation = { package = "astria-sequencer-validation", path = "../astria-sequencer-validation" }
```
So that in the rust code you have the slightly shorter `sequencer_validation::FooBar`. You can see this here with the telemetry crate.

I think that's neater, but I should have probably brought it up for discussion.

File: crates/astria-sequencer-types/src/sequencer_block_data.rs
Comment (SuperFluffy) [inline crates/astria-sequencer-types/src/sequencer_block_data.rs:51]: In PR #376 eyre is removed in favor of typing the errors. Can you do that here as well and merge after that PR?

File: crates/astria-sequencer-types/src/sequencer_block_data.rs
Comment (SuperFluffy) [inline crates/astria-sequencer-types/src/sequencer_block_data.rs:256]: Same here - PR #376 changes this slightly.

**Review by SuperFluffy — APPROVED** [review]
Change looks good. My only asks is to merge this after #376 goes through - and adapt the changes to `astria-sequencet-types` accordingly.

File: crates/astria-sequencer-types/src/sequencer_block_data.rs
Comment (noot) [reply crates/astria-sequencer-types/src/sequencer_block_data.rs:51]: yep, will wait until that's merged!

**Comment by SuperFluffy** [comment]
@noot PR #375 and #376 have been merged, unblocking (but also breaking :-( ) this. If you want I can fix this PR.

**Review by sambukowski — APPROVED** [review]
Other than @SuperFluffy's comments, lgtm

**Comment by noot** [comment]
@SuperFluffy no worries, I can fix it later today!

**Pull request merged**
